### PR TITLE
[SRVKS-1179] enable secret filter informer by default

### PR DESCRIPTION
**Pull request by skonto**
Fixes JIRA #SRVKS-1179

## Proposed Changes

- As per title. We enable it for net-istio and net-kourier. Later on we will add it to net-* manifests too, when we make this the default upstream. 
- Deprecated annotation for net-istio is removed.
- Allows to set back to disabled via the workload overrides in Serving CR as usual.

### CODE REVIEW

File: openshift-knative-operator/pkg/serving/secretinformerfiltering.go [hunk 1]
```diff
@@ -8,18 +8,13 @@ import (
...
 
-const (
-	// TODO: Remove when available in knative.dev/networking/config
-	ServingInternalCertName = "knative-serving-certs"
-	// TODO: Maybe decide to fetch from net-kourier deps instead
-	EnableSecretInformerFilteringByCertUIDEnv = "ENABLE_SECRET_INFORMER_FILTERING_BY_CERT_UID"
-	// TODO: Annotation is deprecated, remove in future releases
-	secretInformerFilteringAnnotation = "serverless.openshift.io/enable-secret-informer-filtering"
-)
+// TODO: Maybe decide to fetch from net-kourier deps instead
```
Comment (skonto) [inline openshift-knative-operator/pkg/serving/secretinformerfiltering.go:16]: I am thinking of moving that const to knative.dev/pkg upstream or something so net-istio could benefit from it. So not eager to bring in the net-kourier dep.

File: openshift-knative-operator/pkg/serving/secretinformerfiltering.go [hunk 2]
```diff
@@ -41,7 +36,8 @@ func enableSecretInformerFilteringTransformers(ks base.KComponent) []mf.Transfor
 
 func injectLabelIntoInternalEncryptionSecret() mf.Transformer {
 	return func(u *unstructured.Unstructured) error {
-		if u.GetKind() == "Secret" && u.GetName() == ServingInternalCertName {
+		//nolint:staticcheck // ignore the deprecation until internal encryption is implemented downstream
```
Comment (skonto) [inline openshift-knative-operator/pkg/serving/secretinformerfiltering.go:39]: As good as it can gets: https://github.com/golangci/golangci-lint/issues/741.

File: openshift-knative-operator/pkg/serving/secretinformerfiltering.go (hunk 1 above)
Comment (ReToCode) [reply openshift-knative-operator/pkg/serving/secretinformerfiltering.go:16]: Yeah, probably better to move it to pkg upstream. Do you want to create a JIRA for that. We'll probably forget the TODO here.

File: openshift-knative-operator/pkg/serving/secretinformerfiltering.go [hunk 3]
```diff
@@ -74,15 +70,10 @@ func configIfUnsetAndCheckIfShouldInject(comp *operatorv1beta1.KnativeServing, d
...
-	// The annotation is deprecated
-	// TODO: Remove this block in future releases
-	if deployment == "net-istio-controller" {
-		if v, ok := comp.GetAnnotations()[secretInformerFilteringAnnotation]; ok {
-			b, _ := strconv.ParseBool(v)
-			// Keep the same behavior as in 1.27
-			return b, common.InjectEnvironmentIntoDeployment("net-istio-controller", "controller",
-				corev1.EnvVar{Name: EnableSecretInformerFilteringByCertUIDEnv, Value: v})
-		}
+	// TODO: remove when set to true at the net-* repos upstream/midstream
```
Comment (ReToCode) [inline openshift-knative-operator/pkg/serving/secretinformerfiltering.go:73]: Same here, JIRA?

File: openshift-knative-operator/pkg/serving/secretinformerfilterting_test.go [hunk 4]
```diff
@@ -185,101 +123,15 @@ func TestSecretInformerFilteringOverride(t *testing.T) {
...
 	}}
 
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
 			tf := enableSecretInformerFilteringTransformers(c.in)
-			if c.shouldAddLabelToSecret {
-				if tf == nil {
-					t.Errorf("Secret transformer should not be nil")
-				}
+			if c.shouldAddLabelToSecret && tf == nil {
```
Comment (ReToCode) [inline openshift-knative-operator/pkg/serving/secretinformerfilterting_test.go:131]: Hm, I'd say the old version was more readable. The else if part is a bit hard to parse.

Comment (skonto) [reply openshift-knative-operator/pkg/serving/secretinformerfilterting_test.go:131]: This part was not tested in the past. We missed that case so it is complete now.

File: openshift-knative-operator/pkg/serving/secretinformerfiltering.go (hunk 3 above)
Comment (skonto) [reply openshift-knative-operator/pkg/serving/secretinformerfiltering.go:73]: See https://issues.redhat.com/browse/SRVKS-1190 and https://issues.redhat.com/browse/SRVKS-1191.

**Comment by skonto** [comment]
/unhold 

Update: docs team has set this as a blocker for 1.32, see https://redhat-internal.slack.com/archives/CGUG69160/p1706703811019039?thread_ts=1706018885.016239&cid=CGUG69160.

**Comment by skonto** [comment]
@ReToCode ready again, had to rebase.

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

/retest-required
